mlrun/feature_store/feature_set.py

```diff
@@ -794,6 +794,21 @@ def to_dataframe(
             time_column=time_column,
             **kwargs,
         )
+        drop_cols = []
+        if target.time_partitioning_granularity:
+            for col in target._legal_time_units:
+                drop_cols.append(col)
+                if col in target.time_partitioning_granularity:
+                    break
+        elif (
+            target.partitioned
+            and not target.partition_cols
+            and not target.key_bucketing_number
+        ):
+            drop_cols = ["year", "month", "day", "hour"]
```

Comment:
This relies on the default partitioning key being `hour`. Is there someplace where this is defined, so we can use the same value and make sure we don't generate a mismatch here?